tools/pyboard.py

```diff
@@ -117,13 +117,32 @@ def inWaiting(self):
             return n_waiting
 
 class Pyboard:
-    def __init__(self, device, baudrate=115200, user='micro', password='python'):
+    def __init__(self, device, baudrate=115200, user='micro', password='python', wait=0):
         if device and device[0].isdigit() and device[-1].isdigit() and device.count('.') == 3:
             # device looks like an IP address
             self.serial = TelnetToSerial(device, user, password, read_timeout=10)
         else:
             import serial
-            self.serial = serial.Serial(device, baudrate=baudrate, interCharTimeout=1)
+            errmsg = '\nFailed to access ' + device
+            if wait:
+                for attempt in range(wait):
+                    try:
+                        self.serial = serial.Serial(device, baudrate=baudrate, interCharTimeout=1)
+                        break
+                    except IOError:
+                        if attempt == 0:
+                            sys.stdout.write('Waiting {} seconds for pyboard '.format(wait))
+                    time.sleep(1)
+                    sys.stdout.write('.')
+                    sys.stdout.flush()
+                else:
+                    raise PyboardError(errmsg)
+                print('')
```

Comment:
This makes pyboard.py print a blank line before the start of the output, which should not happen.